packages/core/core/src/requests/AssetRequest.js

```diff
@@ -0,0 +1,139 @@
+  +type: 'asset_request',
+  run: RunInput => Promise<AssetRequestResult>,
+  input: AssetRequestInput,
+|};
+
+function generateRequestId(type, obj) {
+  return `${type}:${md5FromObject(obj)}`;
+}
+
+export default function createAssetRequest(input: AssetRequestInput) {
+  return {
+    type: 'asset_request',
+    id: getId(input),
+    run,
+    input,
+  };
+}
+
+let handle;
+function getRunTransform(
+  farm: WorkerFarm,
+): TransformationOpts => Promise<{|
+  assets: Array<Asset>,
+  configRequests: Array<{|request: ConfigRequestDesc, result: Config|}>,
+|}> {
+  if (handle != null) {
+    return handle;
+  }
+
+  handle = farm.createHandle('runTransform');
+  return handle;
+  // ? Could this singleton cause problems
+}
+
+const type = 'asset_request';
+
+function getId(input: AssetRequestInput) {
+  // eslint-disable-next-line no-unused-vars
+  let {optionsRef, configRef, ...hashInput} = input;
+  return `${type}:${md5FromObject(hashInput)}`;
+}
+
+async function run({input, api, options, farm}: RunInput) {
+  api.invalidateOnFileUpdate(await options.inputFS.realpath(input.filePath));
+  let start = Date.now();
+  let {configRef, optionsRef, ...request} = input;
+  let {assets, configRequests} = await getRunTransform(farm)({
+    configRef,
+    optionsRef,
+    request,
+  });
+
+  let time = Date.now() - start;
+  for (let asset of assets) {
+    asset.stats.time = time;
+  }
+
+  for (let asset of assets) {
+    for (let filePath of asset.includedFiles.keys()) {
+      api.invalidateOnFileUpdate(filePath);
+      api.invalidateOnFileDelete(filePath);
+    }
+  }
+
+  // Add config requests
+  for (let {request, result} of configRequests) {
+    let id = generateRequestId('config_request', request);
+    await api.runRequest({
+      id,
+      type: 'config_request',
+      run: ({api}) => {
+        if (result.resolvedPath != null) {
+          api.invalidateOnFileUpdate(result.resolvedPath);
+        }
+
+        for (let filePath of result.includedFiles) {
+          api.invalidateOnFileUpdate(filePath);
+        }
+
+        if (result.watchGlob != null) {
+          api.invalidateOnFileCreate(result.watchGlob);
+        }
+
+        if (result.shouldInvalidateOnStartup) {
+          api.invalidateOnStartup();
+        }
+        // ? Better way to make flow happy
+        return Promise.resolve();
+      },
+      input: null,
+    });
+
+    // Add dep version requests
+    for (let [moduleSpecifier] of result.devDeps) {
+      let depVersionRequst = {
+        moduleSpecifier,
+        resolveFrom: result.resolvedPath, // TODO: resolveFrom should be nearest package boundary
+      };
+      let id = generateRequestId('dep_version_request', depVersionRequst);
+      await api.runRequest({
+        id,
+        type: 'version_request',
+        run: ({api}) => {
+          if (options.lockFile != null) {
+            api.invalidateOnFileUpdate(options.lockFile);
+          }
+          // ? Better way to make flow happy
```

Comment:
So, does this request not actually do anything other than invalidate on lock file change? Is it supposed to actually resolve some version?

Reply:
The actual resolution should be happening in the worker, but I forgot to re implement that when we moved config resolution to the workers, so that needs to be fixed. Once that's done it might be beneficial to also store the resulting versions so that we wouldn't have to resolve them again if say a config file changed but the yarn.lock didn't change.